openapi/tests/openapi_integration/test_payload_selector.py

```diff
@@ -0,0 +1,310 @@
+            },
+        }
+    )
+    assert response.ok
+    assert response.json()['result']['points'][0]['payload'] == {
+        "country": {
+            "cities": [
+                {
+                    "name": "Berlin",
+                    "population": 3.7,
+                    "location": {
+                        "lon": 13.76116,
+                        "lat": 52.33826,
+                    },
+                    "sightseeing": ["Brandenburg Gate", "Reichstag"]
+                },
+            ],
+        }
+    }
+
+    # Search with payload selector exclude paths
+    response = request_with_validation(
+        api='/collections/{collection_name}/points/scroll',
+        method="POST",
+        path_params={'collection_name': collection_name},
+        body={
+            "filter": {
+                "must": [
+                    {
+                        "key": "country.name",
+                        "match": {
+                            "value": "Germany",
+                        }
+                    }
+                ]
+            },
+            "limit": 3,
+            "with_payload": {
+                "exclude": ["country.cities"],
+            },
+        }
+    )
+    assert response.ok
+    assert response.json()['result']['points'][0]['payload'] == {
+        "country": {
+            "name": "Germany",
+            "capital": "Berlin",
+        }
+    }
+
+    # Search with payload selector include at array index
+    response = request_with_validation(
+        api='/collections/{collection_name}/points/scroll',
+        method="POST",
+        path_params={'collection_name': collection_name},
+        body={
+            "filter": {
+                "must": [
+                    {
+                        "key": "country.name",
+                        "match": {
+                            "value": "Germany",
+                        }
+                    }
+                ]
+            },
+            "limit": 3,
+            "with_payload": {
+                "exclude": ["country.name", "country.capital", "country.cities[1]"],
+            },
+        }
+    )
+    assert response.ok
+    assert response.json()['result']['points'][0]['payload'] == {
+        "country": {
+            "cities": [
+                {
+                    "name": "Berlin",
+                    "population": 3.7,
+                    "location": {
+                        "lon": 13.76116,
+                        "lat": 52.33826,
+                    },
+                    "sightseeing": ["Brandenburg Gate", "Reichstag"]
+                },
+                {
+                    "name": "Hamburg",
+                    "population": 1.8,
+                    "location": {
+                        "lon": 9.99368,
+                        "lat": 53.55108,
+                    },
+                    "sightseeing": ["Reeperbahn", "Elbphilharmonie"]
+                }
+            ],
+        }
+    }
+
```

Comment:
I understand that this would be very powerful but I am afraid it would add more complexity to the API.
I'd rather not do it until there is very strong case for it.

Reply:
I propose https://github.com/qdrant/qdrant/pull/2756/commits/6c674ff1587d888670009a87bef8fe2d87fce56a

I picked the two important functions to cover get & get_mut.